src/idl_gen_cpp.cpp

```diff
@@ -625,6 +625,15 @@ class CppGenerator : public BaseGenerator {
     return false;
   }
 
+  bool VectorElementUserFacing(const Type& type) const {
+    // Normally, in non-Object-API, we use the non-user-facing type when
+    // emitting the Vector element type, however in the case of enums
+    // we want to avoid that when using scoped-enums.
+    return opts_.g_cpp_std >= cpp::CPP_STD_17 &&
+           opts_.scoped_enums &&
```

Comment:
Instead of `opts_.scoped_enums` we can use `opts_.g_only_fixed_enums` flag.
The `opts.g_only_fixed_enums` which is `(opts_.g_cpp_std >= cpp::CPP_STD_11 || opts_.scoped_enums)`.
Or we should restrict `Array` code generator by `&& opts_.scoped_enums` to be consistent with generated `Vector`:
https://github.com/google/flatbuffers/blob/6400c9b054912ae4761ab3534248b51ffa7a6c1e/src/idl_gen_cpp.cpp#L802-L807

The condition `!opts_.g_only_fixed_enums` for `Arrays` was changed by me in 44bf7198. Before it was (idl_parser.cpp):
```cpp
      // Check if enum arrays are used in C++ without specifying --scoped-enums
      if ((opts.lang_to_generate & IDLOptions::kCpp) && !opts.scoped_enums &&
          IsEnum(subtype)) {
        return Error(
            "--scoped-enums must be enabled to use enum arrays in C++\n");
      }
```
This code is not released yet and we can revert the condition for the `Array` code generator.
@aardappel  need your advice.

Reply:
Not sure that I fully understand what you are recommending... should I just change `scoped_enums` to `g_only_fixed_enums` in the `VectorElementUserFacing` function?

Reply:
I'm not sure yet, but probably yes.

The flag `g_only_fixed_enums` is more generic. If `cpp_std >= C++11` then `g_only_fixed_enums` is set and every generated enum has explicit underlying type independently of user-defined `scoped_enums` flag. This is used when `Array<E, length>` fields are generated.
The `cpp_std==C++17` unconditionally sets `scoped_enums` to true. But in the future, this behavior can be changed. Using  `g_only_fixed_enums` instead of `scoped_enums` for `Vector` will save generated `Vector<E>` fields unchanged.

Reply:
Ok then I will wait for you to get back to me.

Reply:
I am sure that the condition should use the `g_only_fixed_enums` flag:
```cpp
(opts_.g_cpp_std >= cpp::CPP_STD_17) && IsEnum(type) && g_only_fixed_enums;
```
1. The `scoped_enums` implies `g_only_fixed_enums`.
2. The `g_only_fixed_enums` allows to use `Array` with simple enums.

@dpacbach could you adjust this condition in your code?
It doesn't change the logic of this PR because the `scoped_enums` flag implies `g_only_fixed_enums`. It only improves the readability of code. Thanks.

Reply:
@vglavnyy yes we can still change the `Array` code if that's beneficial I think.

Reply:
@vglavnyy Changed `scoped_enums` --> `g_only_fixed_enums`.  Let me know if there is anything else.